plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -185,15 +185,17 @@ public void start(boolean downloadUc) {
         showAllSecurityWarnings();
         bundledPluginVersions = bundledPlugins();
         installedPluginVersions = installedPlugins();
-
-        allPluginsAndDependencies = findPluginsAndDependencies(cfg.getPlugins());
+        List<Exception> exceptions = new ArrayList<>();
+        allPluginsAndDependencies = findPluginsAndDependencies(cfg.getPlugins(), exceptions);
         pluginsToBeDownloaded = findPluginsToDownload(allPluginsAndDependencies);
         effectivePlugins = findEffectivePlugins(pluginsToBeDownloaded);
 
         listPlugins();
         showSpecificSecurityWarnings(pluginsToBeDownloaded);
-        checkVersionCompatibility(jenkinsVersion, pluginsToBeDownloaded);
-
+        checkVersionCompatibility(jenkinsVersion, pluginsToBeDownloaded, exceptions);
+        if (!exceptions.isEmpty()) {
+            throw new AggregatePluginPrerequisitesNotMetException(exceptions);
```

Comment:
can we only output unique results?

```
➜  plugin-installation-manager-tool git:(feature/batch-prerequisites-not-met-exceptions) java -jar plugin-management-cli/target/jenkins-plugin-manager-2.7.2-SNAPSHOT.jar --plugins blueocean --plugins branch-api:2.4.0 -d plugins/ --war ~/Downloads/jenkins.war --credentials abcd:acd:acd
Multiple plugin prerequisites not met:
Plugin blueocean:1.24.4 depends on branch-api:2.6.3, but there is an older version defined on the top level - branch-api:2.4.0,
Plugin blueocean:1.24.4 depends on branch-api:2.6.3, but there is an older version defined on the top level - branch-api:2.4.0,
Plugin blueocean:1.24.4 depends on branch-api:2.6.3, but there is an older version defined on the top level - branch-api:2.4.0,
Plugin blueocean:1.24.4 depends on branch-api:2.6.3, but there is an older version defined on the top level - branch-api:2.4.0
```

Reply:
@timja How I can I reproduce? Which war have you used? This rather indicates that the algorithm should be optimized. Filtering in the exception would hide that flaw IMHO....

Reply:
it's jenkins-2.273, https://github.com/jenkinsci/jenkins/releases/download/jenkins-2.273/jenkins.war

Reply:
With https://github.com/jenkinsci/plugin-installation-manager-tool/pull/298/commits/d2a74719c92584d3b33c98bf86af3bee4e8dc3fd you get
```
Multiple plugin prerequisites not met:
Plugin blueocean:1.24.4 (transitively via blueocean-pipeline-api-impl:1.24.4) depends on branch-api:2.6.3, but there is an older version defined on the top level - branch-api:2.4.0,
Plugin blueocean:1.24.4 (transitively via blueocean-autofavorite:1.2.4) depends on branch-api:2.6.3, but there is an older version defined on the top level - branch-api:2.4.0,
Plugin blueocean:1.24.4 (transitively via blueocean-pipeline-scm-api:1.24.4) depends on branch-api:2.6.3, but there is an older version defined on the top level - branch-api:2.4.0,
Plugin blueocean:1.24.4 (transitively via workflow-multibranch:2.22) depends on branch-api:2.6.3, but there is an older version defined on the top level - branch-api:2.4.0
```

That clarifies that this version is requested from 4 different transitive dependencies.
I would not consolidate to not hide this information from the called.